madara/crates/client/db/src/storage_updates.rs

```diff
@@ -89,7 +89,10 @@ impl MadaraBackend {
     }
 
     pub fn revert_to(&self, revert_to: u64) -> Result<(), MadaraStorageError> {
```

Comment:
you also need to clear the pending block here

Reply:
because we want to be able to keep the pending block around across restarts

I have a PR that will be up soon that replaces the db reads to the pending block to read into RAM, meaning db will only be involved when the pending block changes and needs to be saved again